src/components/media-player.js

```diff
@@ -13,14 +13,33 @@ function MediaPlayer({
   previewMode,
   posterImageUrl,
   extraContentTop,
-  marks
+  marks,
+  onMarkReached,
+  onEndReached,
+  mediaPlayerRef
 }) {
   const trackRef = useRef();
   const [volume, setVolume] = useState(1);
   const [isMuted, setIsMuted] = useState(false);
+  const [lastReachedMark, setLastReachedMark] = useState();
```

Comment:
Not really a big issue, but I think we usually put something as a start value for `useState` (later on you use `null` to indicate absence of data), so maybe we could be more explicit and also initialize it with `null`?